### PR TITLE
GSoC'21 Final Report: Aitik Gupta

### PR DESCRIPTION
**Pull request by aitikgupta**
Scrapping the original idea of keeping two separate blog posts, I've _linked_ the memes (so if anyone's interested, they can actually click on them).

image

### CODE REVIEW

**Comment by davidevaleriani** [comment]
Is this ready to be merged?

**Comment by aitikgupta** [comment]
> Is this ready to be merged?

Let's wait for @story645's review?

File: content/posts/GSoC_2021_Final/index.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+**<ins>Matplotlib: Revisiting Text/Font Handling</ins>**
+
+Here's a [meme](https://user-images.githubusercontent.com/43996118/129448683-bc136398-afeb-40ac-bbb7-0576757baf3c.jpg) I created, to kick things off for this final report!
+## About Matplotlib
+Matplotlib is a comprehensive library for creating static, animated, and interactive visualizations, which has become a _de-facto Python plotting library_.
+
+Much of the implementation behind its font manager is inspired by [W3C](https://www.w3.org/) compliant algorithms, allowing users to interact with font properties like `font-size`, `font-weight`, `font-family`, etc.
+
+#### However, the way Matplotlib handled fonts and general text layout was not ideal, which is what all Summer 2021 was about.
```
Comment (story645) [inline content/posts/GSoC_2021_Final/index.md]:
```suggestion
#### However, the way Matplotlib handled fonts and general text layout was not ideal, which is what Summer 2021 was all about.
```

File: content/posts/GSoC_2021_Final/index.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+### About the Project
+(PS: here's [the link](https://docs.google.com/document/d/11PrXKjMHhl0rcQB4p_W9JY_AbPCkYuoTT0t85937nB0/) to my GSoC proposal, if you're interested)
+
+Overall, the project was divided into two major subgoals:
+1. Font Subsetting
+2. Font Fallback
+
+But before we take each of them on, we should get an idea about some basic terminology for fonts (which are a _lot_, and are rightly _confusing_)
+
+The [PR: Clarify/Improve docs on family-names vs generic-families](https://github.com/matplotlib/matplotlib/pull/20346/files) brings about a bit of clarity about some of these terms. The next section has a linked PR which also explains the types of fonts and how is that relevant to Matplotlib.
```
Comment (story645) [inline content/posts/GSoC_2021_Final/index.md]:
```suggestion
The [PR: Clarify/Improve docs on family-names vs generic-families](https://github.com/matplotlib/matplotlib/pull/20346/files) brings about a bit of clarity about some of these terms. The next section has a linked PR which also explains the types of fonts and how that is relevant to Matplotlib.

```

File: content/posts/GSoC_2021_Final/index.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+| Nov 2, 2020  | [Expand ScalarMappable.set_array to accept array-like inputs](https://github.com/matplotlib/matplotlib/pull/18870)      | (+28 −4)        | MERGED |
+| Nov 8, 2020  | [Add overset and underset support for mathtext](https://github.com/matplotlib/matplotlib/pull/18916)                    | (+71 −0)        | MERGED |
+| Nov 14, 2020 | [Strictly increasing check with test coverage for streamplot grid](https://github.com/matplotlib/matplotlib/pull/18947) | (+54 −2)        | MERGED |
+| Jan 11, 2021 | [WIP: Add support to edit subplot configurations via textbox](https://github.com/matplotlib/matplotlib/pull/19271)      | (+51 −11)       | DRAFT  |
+| Jan 18, 2021 | [Fix over/under mathtext symbols](https://github.com/matplotlib/matplotlib/pull/19314)                                  | (+7,459 −4,169) | MERGED |
+| Feb 11, 2021 | [Add overset/underset whatsnew entry](https://github.com/matplotlib/matplotlib/pull/19497)                              | (+28 −17)       | MERGED |
+| May 15, 2021 | [Warn user when mathtext font is used for ticks](https://github.com/matplotlib/matplotlib/pull/20235)                   | (+28 −0)        | MERGED |
+
```
Comment (story645) [inline content/posts/GSoC_2021_Final/index.md:145]: Can you add a small bullet list summarizing the GSOC 2021 PRs? As a done/in progress/to do?

File: content/posts/GSoC_2021_Final/index.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+## Acknowledgements
```
Comment (story645) [inline content/posts/GSoC_2021_Final/index.md:147]: What I love most about this (❤️) is that it breaks down that each person contributes something different. (Also I totally learn from Antony in the same way).

Comment (aitikgupta) [reply content/posts/GSoC_2021_Final/index.md:147]: Ahaha ikr!

**Review by story645 — APPROVED** [review]

**Pull request merged**